pkg/storage/engine/mvcc.go

```diff
@@ -1852,6 +1852,23 @@ func mvccResolveWriteIntent(
 	// For cases where there's no write intent to resolve, or one exists
 	// which we can't resolve, this is a noop.
 	if !ok || meta.Txn == nil || !roachpb.TxnIDEqual(intent.Txn.ID, meta.Txn.ID) {
```

Comment:
Your guesses are correct. `mvccGetUsingIter` should return without an error even when `!ok`. It will hit the `v == nil` case.